### PR TITLE
Support multiple dsoType values in discovery

### PR DESCRIPTION
**Pull request by benbosman**
## References
* Related to [REST Contract](https://github.com/DSpace/Rest7Contract/pull/146)

## Description
This is a minor change to support e.g. /server/api/discover/search/objects?dsoType=Collection&dsoType=Community to return communities and collections

We discovered this was relevant to create the Angular UI to import or export a CSV, where a collection or community should be selected (and both should be selectable)

## Instructions for Reviewers
There's no need to reindex data. You can test it using:
* no dsoType parameter
* /server/api/discover/search/objects?dsoType=Collection&dsoType=Community
* /server/api/discover/search/objects?dsoType=Item
* /server/api/discover/search/objects?dsoType=Item&dsoType=Collection
* …

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/DiscoverQueryBuilder.java
Comment (tdonohue) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/DiscoverQueryBuilder.java:83]: Could you please add Javadocs to this new method?

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/DiscoverQueryBuilder.java
Comment (tdonohue) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/DiscoverQueryBuilder.java:128]: Again, please add Javadocs to this new method.

**Review by tdonohue — APPROVED** [review]
👍 Looks good to me overall, @benbosman !  I've tested it manually and it works.

A few very minor requests before we merge...I'd just like us to add JavaDocs to the new methods.  Beyond that, this will be ready to merge once JavaDocs are added.

**Comment by benbosman** [comment]
@tdonohue Samuel has added the JavaDocs

**Review by tdonohue — APPROVED** [review]
Thanks @benbosman and @samuelcambien ! Merging now

**Pull request merged**
